Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.cpp

```diff
@@ -32,10 +32,10 @@
 
 namespace WebKit {
 
-void CGDisplayList::encode(IPC::Encoder& encoder) const
+void CGDisplayList::encode(IPC::Encoder& encoder) &&
 {
     encoder << m_displayList;
-    encoder << m_surfaces;
+    encoder << WTFMove(m_surfaces);
```

Comment:
@hortont424 is it correct to move `m_surfaces` here? I tried it, it seems to work, but wanted to check with you prior to merging

Reply:
Encoding something shouldn't mutate its state, right?

Reply:
I think it's correct in this case, the type of `this` is an r-value reference, so the caller has promised us that the object won't be used again, and we can consume its contents as we desire.

Ultimately the innermost type here (`MachSendRight`) is actually consumed by writing it to IPC (the kernel transfers ownership of the object to the new process), so propagating that up into wrapping types makes sense (or they can make an explicit copy of MachSendRights if they want).